data/src/main/java/org/apache/iceberg/data/BaseFileWriterFactory.java

```diff
@@ -118,7 +118,7 @@ public DataWriter<T> newDataWriter(
 
         case PARQUET:
           Parquet.DataWriteBuilder parquetBuilder =
-              Parquet.writeData(outputFile)
+              Parquet.writeData(file)
```

Comment:
Since `encryptingOutputFile` will create a key and an `AesGcmOutputFile`, I don't think it should be called unless it is going to be used. I think `outputFile` should be removed and the branches that pass an `OutputFile` (ORC and Avro) should call `file.encryptingOutputFile()` inline.

Reply:
SGTM